projects/04_cube/Cube.cpp

```diff
@@ -220,19 +244,22 @@ void CubeApp::Render()
     // Build command buffer
     PPX_CHECKED_CALL(frame.cmd->Begin());
     {
-        grfx::RenderPassPtr renderPass = swapchain->GetRenderPass(imageIndex);
+        grfx::RenderPassPtr renderPass = mRenderPasses[imageIndex];
         PPX_ASSERT_MSG(!renderPass.IsNull(), "render pass object is null");
 
         grfx::RenderPassBeginInfo beginInfo = {};
         beginInfo.pRenderPass               = renderPass;
         beginInfo.renderArea                = renderPass->GetRenderArea();
-        beginInfo.RTVClearCount             = 1;
-        beginInfo.RTVClearValues[0]         = {{0, 0, 0, 0}};
-        beginInfo.DSVClearValue             = {1.0f, 0xFF};
+
+        // Clear RTV to greyish blue
+        grfx::RenderTargetClearValue rtvClearValue = {0.23f, 0.23f, 0.33f, 0};
+        grfx::DepthStencilClearValue dsvClearValue = {1.0f, 0xFF};
 
         frame.cmd->TransitionImageLayout(renderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PRESENT, grfx::RESOURCE_STATE_RENDER_TARGET);
         frame.cmd->BeginRenderPass(&beginInfo);
         {
+            frame.cmd->ClearRenderTarget(renderPass->GetRenderTargetImage(0), rtvClearValue);
```

Comment:
why do we move to explicit clear instead of doing the clear in the beginrenderpass?

Reply:
The commit message says that this change to the sample was to test the implementation of the clear functions.